src/hera/workflows/io/_io_mixins.py

```diff
@@ -166,39 +145,16 @@ def _get_as_arguments(self) -> ModelArguments:
             # If it is a templated string, it will be unaffected as `"{{mystr}}" == serialize("{{mystr}}")``
             templated_value = serialize(self_dict[field])
 
-            if get_origin(annotations[field]) is Annotated:
-                annotation = get_args(annotations[field])[1]
-                if isinstance(annotation, Parameter) and annotation.name:
-                    params.append(ModelParameter(name=annotation.name, value=templated_value))
-                elif isinstance(annotation, Artifact) and annotation.name:
-                    artifacts.append(ModelArtifact(name=annotation.name, from_=templated_value))
+            if (param := get_annotated_metadata(annotations[field], Parameter)) and param.name:
+                params.append(ModelParameter(name=param.name, value=templated_value))
+            elif (artifact := get_annotated_metadata(annotations[field], Artifact)) and artifact.name:
+                artifacts.append(ModelArtifact(name=artifact.name, from_=templated_value))
             else:
                 params.append(ModelParameter(name=field, value=templated_value))
```

Comment:
Is this definitely correct? I'm surprised we want to create a `ModelParameter` if something is annotated as an `Artifact` but happens to be missing the `name`. Same question for the code below.

Reply:
I'm not 100% sure it is correct behavior, but found that this also changes the behavior. Maybe we don't have test cases for not having parameter or artifact (or its name) while annotated. I'll update this to act like before, and will follow up in next pr.

Reply:
It's not really correct but I want to revisit name-related script runner stuff for some bugfixes/enhancements, see https://github.com/argoproj-labs/hera/issues/1165. It would be in non-`io_mixin` code too so better as a separate PR